File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
from itertools import product

def solution(word):
    answer = 0
    dict = [] 
    vowel = ['A', 'E', 'I', 'O', 'U']
    
    for i in range(1,6):
        for v in product(vowel,repeat=i):
            dict.append(''.join(v))
    dict.sort()
    answer = dict.index(word) + 1
    
    return answer

File: 프로그래머스/2/84512. 모음 사전/모음 사전.java
import java.util.*;

class Solution {
    
    static char[] chars = {'A','E','I','O','U'};
    static String answer;
    static int count;
    
    public int solution(String word) {
        answer = word;
        count = 0;
        dfs("");
        return count -1 ;
    }
    
    private boolean dfs(String current) {
        
        count++;
        
        if (current.equals(answer)) { 
            return true;
        }
        
        if (current.length() == 5) { 
            return false;
        }
        
        for (int i = 0; i < chars.length; i++) {
            if (dfs(current + chars[i])) {
                return true;
            }
        }
        
        return false;
    }
}